### PR TITLE
add tests, make golint pass

### PR DESCRIPTION
**Pull request by b5**
might as well write tests while learning stuff. I've also added comments to make golint happy, feel free to let me know if any of those are inaccurate.

cc @Stebalien

### CODE REVIEW

File: coding.go [hunk 1]
```diff
@@ -43,12 +44,13 @@ func (d *safeBlockDecoder) Decode(block blocks.Block) (Node, error) {
...
-	} else {
-		// TODO: get the *long* name for this format
-		return nil, fmt.Errorf("unrecognized object type: %d", ty)
 	}
+
+	// TODO: get the *long* name for this format
+	return nil, fmt.Errorf("unrecognized object type: %d", ty)
 }
 
+// DefaultBlockDecoder is a safeBlockDecoder with no decoders registered
```
Comment (Stebalien) [inline coding.go]: `safeBlockDecoder` isn't exported so I wouldn't mention it by here. I'd just say "DefaultBlockDecoder is the block decoder used by `Decode`" and maybe mention that (a) it's threadsafe and (b) users should register block decoders with it.

File: format.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 	cid "github.com/ipfs/go-cid"
 )
 
+// Resolver is the interface that details paths a node contains
```
Comment (Stebalien) [inline format.go]: `Resolver` is more something that can be resolved through than something that "details" (although I'm planning on reworking these interfaces in the near future).

File: format.go [hunk 3]
```diff
@@ -19,6 +20,7 @@ type Resolver interface {
 	Tree(path string, depth int) []string
 }
 
+// Node is an object in a Merkle DAG graph
```
Comment (Stebalien) [inline format.go]: * DAG graph is redundant.
* We usually say merkledag as one word.

File: format.go [hunk 4]
```diff
@@ -40,7 +42,9 @@ type Node interface {
 	Size() (uint64, error)
 }
 
+// NodeGetter is the interface that wraps the basic get method
```
Comment (Stebalien) [inline format.go]: NodeGetter is really just a read-only DAG service. I'd take a look at #8, it makes a lot of this clearer. Actually, I recommend making the pull request against that branch as I'll likely merge that first.

File: format.go [hunk 5]
```diff
@@ -56,6 +60,11 @@ type Link struct {
 	Cid *cid.Cid
 }
 
+// GetNode returns the MDAG Node that this link points to
```
Comment (Stebalien) [inline format.go]: IPLD Node.

**Comment by b5** [comment]
Oooooh ok makes sense. Well at least I still learned a thing :)

**Comment by Stebalien** [comment]
Also, documentation is *always* welcome.

**Pull request closed without merging**